### PR TITLE
WSTEAMA-1110 - Add VJ Embed on Home and Topic pages

### PR DESCRIPTION
**Pull request by louisearchibald**
Resolves JIRA [[1110](https://jira.dev.bbc.co.uk/browse/WSTEAMA-1110)]

Overall changes
======
Allows an embed component to be rendered on Home and Topic pages when the embed html is present inside of the curation data.

Code changes
======

- Update to hindi homepage fixture data to include the vj curation.
- Update to Curation props to include an embed object.
- Embed object passed to Curation from Home page and Topic page.
- Adds a constant for `EMBED` inside of getComponentName and a conditional case added to return it when present.
- Test case added to cater for the presence of an `EMBED` component.

Testing
======
1. Checkout to the branch and `yarn dev` to spin up a local server.
2. Navigate to [http://localhost:7080/hindi](http://localhost:7080/hindi) and the election banner should render in the first position on the page. (or in the first position underneath an ad if you have ads switched on locally).

Screenshot 2024-03-21 at 12 39 49

**Note**: Only home page testing notes are listed above, however I have tested the banner renders on a Topic page by inserting the embed curation into a topic fixture and spinning up a `yarn dev` to view it on the page. Future BFF work in an upcoming sprint will allow us to test this more throughly when Topic pages are able to support the same curations as Home pages.

### CODE REVIEW

File: src/app/components/Curation/getComponentName/index.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ export const COMPONENT_NAMES = {
   NOT_SUPPORTED: 'not-supported',
   MOST_READ: 'most-read',
   RADIO_SCHEDULE: 'radio-schedule',
+  EMBED: 'vj-embed',
```
Comment (karinathomasbbc) [inline src/app/components/Curation/getComponentName/index.ts]: Do we need to include `vj` in here? As there might be other types of Embed in future (not just VJ ones)

```suggestion
  EMBED: 'embed',
```

File: src/app/components/Curation/index.tsx [hunk 2]
```diff
@@ -49,21 +51,23 @@ export default ({
...
+    embed,
   });
 
   const GridComponent = getGridComponent(componentName);
 
   const isFirstCuration = position === 0;
   const curationSubheading = title || topStoriesTitle;
   const id = idSanitiser(curationSubheading);
-
+  const vjEmbed = embed;
```
Comment (karinathomasbbc) [inline src/app/components/Curation/index.tsx]: I think we can delete this

File: src/app/components/Curation/index.tsx [hunk 3]
```diff
@@ -91,6 +95,8 @@ export default ({
       );
     case RADIO_SCHEDULE:
       return <RadioSchedule initialData={radioSchedule} />;
+    case EMBED:
+      return vjEmbed ? <Embed oembed={embed} type="" /> : null;
```
Comment (karinathomasbbc) [inline src/app/components/Curation/index.tsx]: Do we need ensure that there is an embed before we can render the component? Also, is type a mandatory prop here? If so, is there another value other than `""` that we can use?

```suggestion
      return <Embed oembed={embed} type="" />;
```

File: src/app/pages/HomePage/HomePage.tsx [hunk 4]
```diff
@@ -34,6 +34,8 @@ export interface HomePageProps {
   };
 }
 
+//test
```
Comment (karinathomasbbc) [inline src/app/pages/HomePage/HomePage.tsx]: Let's delete this 😸

File: src/app/pages/HomePage/HomePage.tsx [hunk 5]
```diff
@@ -118,8 +120,9 @@ const HomePage = ({ pageData }: HomePageProps) => {
                       nthCurationByStyleAndProminence={
                         nthCurationByStyleAndProminence
                       }
+                      embed={embed}
                     />
-                    {index === 0 && <MPU />}
+                    {index === 1 && <MPU />}
```
Comment (karinathomasbbc) [inline src/app/pages/HomePage/HomePage.tsx]: We need to have a think about this one - let's switch ads on on localhost, and we can show to Gavin / UX. The issue here is that we will have:
- ads banner (or grey placeholder)
- election banner
- another ads banner
- top stories

There might be a concern that users will have to do a lot of scrolling before they see content...

File: src/app/components/Ad/index.tsx [hunk 6]
```diff
@@ -1,29 +1,26 @@
...
 import { RequestContext } from '#contexts/RequestContext';
 import AmpAd from './Amp';
 import CanonicalAd from './Canonical';
 import { AdProps } from './types';
 import AdBootstrapJs from './Canonical/AdBootstrapJs';
 
 const AdContainer = ({ slotType, className, adcampaign }: AdProps) => {
-  const { isAmp, showAdsBasedOnLocation } = useContext(RequestContext);
-  const { enabled: adsEnabled } = useToggle('ads');
+  const { isAmp } = useContext(RequestContext);
```
Comment (karinathomasbbc) [inline src/app/components/Ad/index.tsx]: We need to revert these changes - this was just to display the ad placeholder in Storybook in the spike, but now we don't need this.

**Review by karinathomasbbc — APPROVED** [review]
Looks good to me!

File: src/app/pages/HomePage/HomePage.tsx (hunk 5 above)
Comment (karinathomasbbc) [reply src/app/pages/HomePage/HomePage.tsx]: Although this change has been reverted, I would like to have do a show & tell with Product & UX with ads enabled, so that we see what this actually looks like

File: src/app/components/Curation/index.tsx (hunk 3 above)
Comment (louisearchibald) [reply src/app/components/Curation/index.tsx]: Oooh yes I had forgotten to ask about the "", I was unsure what value to insert at the time but it is a mandatory prop so I'd popped in the empty string as a temp workaround in order to check my code was working 😅 so I'll have to change this still, thank you for the reminder.

I originally had `return <Embed oembed={embed} type="" />` but it cause oembed to scream about a type error I was unable to solve at the time, so I can revisit this if we'd rather not have the check in there first.

Comment (karinathomasbbc) [reply src/app/components/Curation/index.tsx]: It's OK if we still need the check - especially if TS was screaming at you! 😱 And if an empty type is OK, then let's leave that too 😸

File: data/kyrgyz/homePage/index.json
Comment (karinathomasbbc) [inline data/kyrgyz/homePage/index.json:1]: Let's revert this file, as that would fix the tests. I think we probably did this on the original spike because at the time there was only a Kyrgyz homepage available on Storybook..., but now we can view all homepages, so this is no longer needed.

File: src/app/models/types/curationData.ts [hunk 7]
```diff
@@ -56,6 +56,9 @@ export interface BaseCuration {
   curationType?: string;
   mostRead?: MostReadData;
   radioSchedule?: RadioScheduleData[];
+  embed?: {
+    html: string;
+  };
```
Comment (karinathomasbbc) [inline src/app/models/types/curationData.ts]: Do we maybe want to use the `OEmbedData` type from https://github.com/bbc/simorgh/blob/e2abf581b77d21d9309720e348f7fee2e85416b8/src/app/components/Embeds/types.ts#L9?

File: src/app/components/Curation/index.tsx [hunk 8]
```diff
@@ -51,19 +53,21 @@ export default ({
...
+    embed,
   });
 
   const GridComponent = getGridComponent(componentName);
 
   const isFirstCuration = position === 0;
   const curationSubheading = title || topStoriesTitle;
   const id = idSanitiser(curationSubheading);
-
+  const vjEmbed = embed;
```
Comment (amoore108) [inline src/app/components/Curation/index.tsx]: Can `embed` be used instead of reassigning to `vjEmbed`?

File: src/app/components/Curation/index.tsx (hunk 3 above)
Comment (amoore108) [inline src/app/components/Curation/index.tsx:99]: Think we could probably just return `<Embed oembed={embed} type="" />` rather than having a ternary. The `Embed` component itself should return `null` if there is no html for it to render.

File: src/app/components/Curation/index.tsx (hunk 8 above)
Comment (louisearchibald) [reply src/app/components/Curation/index.tsx]: Yes, changed. Thanks @amoore108 😄

File: src/app/components/Curation/index.tsx [hunk 9]
```diff
@@ -91,6 +95,8 @@ export default ({
       );
     case RADIO_SCHEDULE:
       return <RadioSchedule initialData={radioSchedule} />;
+    case EMBED:
+      return embed ? <Embed oembed={embed} type="" /> : null;
```
Comment (karinathomasbbc) [inline src/app/components/Curation/index.tsx]: 👋 I was having a wee poke around in the OEmbed code, and I couldn't seem to find this `type` parameter being used. Perhaps a question for @amoore108, but would it be OK if we were to make `type` optional on [OembedProps](https://github.com/bbc/simorgh/blob/latest/src/app/components/Embeds/types.ts#L27-L30), so that we could remove it and not hardcode it to `""` here?

Comment (amoore108) [reply src/app/components/Curation/index.tsx]: Yea I had a play around with this too. I think `type` can be removed from `OEmbedProps`, its not used in the components that its typing.

This has a knock-on effect for the Flourish embeds tests: https://github.com/bbc/simorgh/blob/dbd29e743b43da179278c5c2b3733780fba093e7/src/app/components/Embeds/FlourishEmbed/index.test.tsx#L9-L16

☝️ They aren't actually typed correctly 😱 . They should be using `OEmbedData` instead, so if you change it to that it should help. You may need to use `} satisfies OEmbedData` to get it working.

I don't mind committing those changes if needed?

**Review by LilyL0u — APPROVED** [review]
Checked it on the page too, it's big and its bold and it is THERE 🔥

File: src/app/components/Curation/index.tsx (hunk 9 above)
Comment (louisearchibald) [reply src/app/components/Curation/index.tsx]: @amoore108 @karinathomasbbc I've made the changes suggested - I had to update the width & height props on `OEmbedData` to support a string and number to keep the flourish tests happy as the iframe supports both of these types. Hope this is okay?

Comment (amoore108) [reply src/app/components/Curation/index.tsx]: Yeah that's perfect, thanks for fixing! 🙌🏻
